plugins/org.yakindu.base.expressions/src/org/yakindu/base/expressions/linking/LinkingDiagnosticMessageProvider.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2018 committers of YAKINDU and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * Contributors:
+ * 	committers of YAKINDU - initial API and implementation
+ * 
+ */
+package org.yakindu.base.expressions.linking;
+
+import org.eclipse.xtext.diagnostics.Diagnostic;
+import org.eclipse.xtext.diagnostics.DiagnosticMessage;
+import org.eclipse.xtext.diagnostics.Severity;
+import org.eclipse.xtext.linking.impl.IllegalNodeException;
+
+public class LinkingDiagnosticMessageProvider extends org.eclipse.xtext.linking.impl.LinkingDiagnosticMessageProvider {
+	@Override
+	public DiagnosticMessage getUnresolvedProxyMessage(ILinkingDiagnosticContext context) {
+		String linkText = "";
+		try {
+			linkText = context.getLinkText();
+		} catch (IllegalNodeException e) {
+			linkText = e.getNode().getText();
+		}
+		String message = "Could not find declaration of '" + linkText + "'";
```

Comment:
Now, you are missing the information that is stored in `referenceType.getName()`. For example, `var x : noType` previously gave "Couldn't resolve reference to Type noType". Now it just says "Could not find declaration of noType". 

Another example: `[active(noSuchState)] ` previously gave "Couldn't resolve reference to State 'noSuchState'", now it just says "Could not find declaration of noSuchState''.